### DISCUSSION TITLE
How to load a yaml-var from a file by "fly ... set-pipeline ... -l var-file"?

### DISCUSSION THREAD

**Discussion by onlookor** [concourse/concourse#8372] [Help & Support]
### Intention
I have some command looks like this in my task config:

> run:
           path: xsltproc
           args: ["--xinclude","--stringparam","-o","dist/html/","stylesheets/custom/chunk.xsl","docbook_source/test.docbk"]
>

The length of args is not fixed yet, so I try to take a var inplace:

>  run:
        path: xsltproc
        args: ((xsltproc_args))
>

And a line in vars.yml:

> xsltproc_args: ["--xinclude","--stringparam","-o","dist/html/","stylesheets/custom/chunk.xsl","docbook_source/test.docbk"]

Put together:

> fly -t test sp -p xslt -c mypipeline.yml -l vars.yml

### Problem

This task ends with a message:

>failed to create task config from bytes ci/task/xsltproc.yml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go struct field TaskRunConfig.run.args of type []string

Looks like concourse takes the line "xsltproc_args: [...]" in vars.yml as a string value variable. So I try it with "-y":

> fly -t test sp -p xslt -c mypipeline.yml -l vars.yml -y xsltproc_args=["--xinclude", ... ]

The shell complains:

> zsh: no matches found: xsltproc_args_html=[--xinclude,--stringparam, ... 

### My question

So, how should I load a yaml array value from a vars file? Or, any way getting around it to use variable length args of "run" in a task config?

**Comment by klas-s** [comment]
I was also unsuccessful with the above, so you're not doing anything obviously wrong. Although I think you're using the `-y` incorrectly, as that is supposed to be used with a yaml document, not a `variable=value` as you seem to have attempted. Might be variables, as implemented, do not support this scenario.

Two other options that come to mind is either building the list of arguments in a step of your job, and then just running that in your task, which is now "static". Or you could resort to something like ytt for some serious yaml templating. The latter approach is used in Concourse CI pipeline that builds the resource types. See e.g. 
https://github.com/concourse/ci/blob/0d54e5ad00118e648f47359e66b9bbc2c1bbcfa1/pipelines/resources/template.yml

I'd probably go with just creating a list of variable arguments in a separate build step, and putting that into a file, and then using it in a subsequent build step that actually runs xsltproc. I, personally, think Concoures pipelines should not attempt to handle every single piece of variation with switches and such. There is shell, Ruby, Python, etc, for that.
